sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/TestUtilities.scala

```diff
@@ -0,0 +1,63 @@
+package scala.scalanative
+package sbtplugin
+
+import scala.scalanative.test.{FakeFingerprint, NativeFingerprint}
+
+import sbt._
+import sbt.testing._
+
+object TestUtilities {
+
+  def remapTestDefinition(test: TestDefinition): TestDefinition = {
+    val newFingerprint = new FakeFingerprint(test.fingerprint)
+    new TestDefinition(test.name,
+                       newFingerprint,
+                       test.explicitlySpecified,
+                       test.selectors)
+  }
+
+  def createTestMain(tests: Seq[TestDefinition]): String = {
+    val cases  = tests.map(createTestCase)
+    val indent = "    "
+
+    s"""|object TestMain {
+        |  def main(args: Array[String]): Unit = args match {
+        |    ${cases.mkString(System.lineSeparator + indent)}
+        |    case Array() =>
+        |      ()
+        |  }
+        |}""".stripMargin
+  }
+
+  private def createTestCase(test: TestDefinition): String = {
+    val body =
+      test.fingerprint match {
```

Comment:
Will this work for other frameworks in the future like ScalaTest/ScalaCheck/etc?